File: week5-database/morn_exe/quiz/scrabble_dt_sol1.rb
# inject solution. .reduce and .inject is the same

class Scrabble

  def self.score(word)

    lookup = {
      'a' => 1,
      'b' => 3,
      'c' => 3
    }

    total_score = 0;

    word.split('').each do |char|
      total_score = total_score + lookup[char]
    end

    total_score

    # word.split('').reduce(0) do |total, char|
    #   total = total + lookup[char]
    #   total
    # end
  end

end

File: week5-database/morn_exe/quiz/scrabble_dt_sol1.js
var Scrabble = {

  score: function (word) {

    var lookup = {
      'a': 1,
      'b': 3,
      'c': 3
    }

    var chars = word.split('');
    var total = 0;

    for(var i=0; i<chars.length; i++) {
      total = total + lookup[chars[i]];
    }

    return total;
  }

}